pkg/pub_package_reader/test/integration_test.dart

```diff
@@ -66,5 +66,36 @@ dev_dependencies:
         )),
       );
     });
+
+    test('malformatted version', () async {
```

Comment:
Should we put something like this into our test data in the future?

Reply:
Any `ArchiveIssue` that we emit here is an upload-blocker, and we are testing this via full filesystem-based parsing. Nevertheless, we should make an integration test that tries to upload _dummy_pkg with the wrong internals.

Reply:
Although... I guess at one point `pub` client itself will block the upload, so the usefulness of that test is probably only for a short time-window.